code/scripting/scripting.h

```diff
@@ -243,6 +242,10 @@ class script_state
 	void ParseGlobalChunk(ConditionalActions hookType, const char* debug_str=nullptr);
 	bool ParseCondition(const char *filename="<Unknown>");
 	void AddConditionedHook(ConditionedHook hook);
+	// ActiveActions lets code that might run scripting hooks know whether any scripts are even registered for it.
+	// AssayActions is responsible for keeping it up to date.
+	bool ActiveActions[ConditionalActions::CHA_LAST+1];
```

Comment:
The check if there is an active action should be put into a function which can then use this data structure internally. That will make it easier to refactor the internals of the scripting system later if that is necessary.

Reply:
I've refactored everything to do it this way.